h/services/auth_ticket.py

```diff
@@ -62,68 +56,32 @@ def verify_cookie(self):
 
         return self._user
 
-    def create_cookie(self, userid):
-        """
-        Create headers for a persistent cookie to log in the user.
-
-        :param userid: Id of the user to log in
-        :return: An iterable of headers to return to the browser
-        """
+    def add_ticket(self, userid: str, ticket_id: str) -> None:
+        """Add a new auth ticket for the given userid and token_id to the DB."""
 
         # Update the user cache to allow quick checking if we are called again
         self._user = self._user_service.fetch(userid)
         if self._user is None:
             raise ValueError(f"Cannot find user with userid {userid}")
 
         ticket = AuthTicket(
-            id=base64.urlsafe_b64encode(os.urandom(32)).rstrip(b"=").decode("ascii"),
+            id=ticket_id,
             user=self._user,
             user_userid=self._user.userid,
             expires=datetime.utcnow() + self.TICKET_TTL,
         )
         self._session.add(ticket)
 
-        return self._cookie.get_headers([self._user.userid, ticket.id])
+    def remove_ticket(self, ticket_id: str) -> None:
+        """Remove any ticket with the given ID from the DB."""
 
-    def revoke_cookie(self):
-        """
-        Create headers to revoke the cookie used to log in a user.
+        self._session.query(AuthTicket).filter_by(id=ticket_id).delete()
 
-        :return: An iterable of headers to return to the browser
-        """
-
-        _, ticket_id = self._get_cookie_value()
-        if ticket_id:
-            self._session.query(AuthTicket).filter_by(id=ticket_id).delete()
-
-        # Empty the cached user to force revalidation
+        # Empty the cached user to force revalidation.
         self._user = None
 
-        return self._cookie.get_headers(None, max_age=0)
-
-    def _get_cookie_value(self):
-        value = self._cookie.get_value()
-        if not value:
-            return None, None
-
-        return value
-
 
 def factory(_context, request):
-    """Return a AuthCookieService instance for the passed context and request."""
-
-    cookie = SignedCookieProfile(
-        # This value is set in `h.auth` at the moment
-        secret=request.registry.settings["h_auth_cookie_secret"],
-        salt="authsanity",
-        cookie_name="auth",
-        max_age=30 * 24 * 3600,  # 30 days
-        httponly=True,
-        secure=request.scheme == "https",
-    )
```

Comment:
The cookie profile was previously defined here in `AuthCookieService`'s service factory. This meant that if multiple security policies were using `AuthCookieService` (e.g. `CookiePolicy` and `APICookiePolicy`) each policy had to share the same cookie, since the cookie name and other settings were passed once to the per-request-singleton `AuthCookieService`'s `__init__()` method.

This also meant that `AuthCookieService` could not be used by any security policy that was not based on cookies (e.g. a header-based policy) whereas the new `AuthTicketService` knows only about authtickets in the DB and is agnostic about how authentication is handled in requests and responses.